firefox_puppeteer/ui/toolbars.py

```diff
@@ -0,0 +1,337 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import Wait, By
+
+from ..api.keys import Keys
+from ..api.l10n import L10n
+from ..base import BaseLib
+from ..decorators import use_class_as_property
+
+
+class NavBar(BaseLib):
+
+    @use_class_as_property('ui.toolbars.LocationBar')
+    def locationbar(self):
+        """Provides utility members for accessing and manipulationg the
+        locationbar.
+
+        See the :class:`~ui.toolbars.LocationBar` reference.
+        """
+
+    @property
+    def back_button(self):
+        """
+        :returns: The back button element.
+        """
+        return self.marionette.find_element('id', 'back-button')
+
+    @property
+    def forward_button(self):
+        """
+        :returns: The forward button element.
+        """
+        return self.marionette.find_element('id', 'forward-button')
+
+    @property
+    def home_button(self):
+        """
+        :returns: The home button element.
+        """
+        return self.marionette.find_element('id', 'home-button')
+
+    @property
+    def menu_button(self):
+        """
+        :returns: The menu button element.
+        """
+        return self.marionette.find_element('id', 'PanelUI-menu-button')
+
+
+class LocationBar(BaseLib):
+    """Various utilities for interacting with the location bar (the text area
+    of the ui that typically displays the current url).
+    """
+
+    dtds = ["chrome://branding/locale/brand.dtd",
+            "chrome://browser/locale/browser.dtd"]
```

Comment:
We should not redefine all the DTDs in each class. Instead we should pass through a BaseWindow reference to each UI class. I'm happy to move this out to later but really file a new bug for it.